WordPress/gutenberg
# Social Icons: Add icon & background color options

Fixes: https://github.com/WordPress/gutenberg/issues/21605

## Description

This is an alternate approach to https://github.com/WordPress/gutenberg/pull/25372 using CSS variables on the primary Social Links block that are then used by styles on the individual Social Link blocks. 

Compared to the original approach this PR:
- Avoids having to maintain and pass color hex values through context to the individual blocks
- Does not need to manually adjust server side rendering of social link blocks
- Social Links block gets icon and background colors set as CSS variables which are used by CSS rules targeting the individual Social Link blocks.

## How has this been tested?

Manual Testing.

1. Create new post and add social links block with a few links added to it.
2. Save post
3. Checkout this PR
4. Reload the editor and ensure no block validation errors
5. Select the social links block and toggle icon or background color options
6. Try switching block styles
    * "Logos only" style will clear any previous background color selection
    * Having only an icon color selected will result in no icon background color
7. Save the post and confirm colors on the frontend match the editor.

## Screenshots
| TwentyTwentyOne | TwentyTwenty |
|---|---|
| ![SocialLinksCSSVariables](https://user-images.githubusercontent.com/60436221/104686498-0e5ae880-5749-11eb-9a62-c3426cc1600d.gif) | ![SocialLinksCSSVariables2](https://user-images.githubusercontent.com/60436221/104686472-026f2680-5749-11eb-82e2-b0f3fba1ab9e.gif) |

## Types of changes
Enhancement

## Checklist:
